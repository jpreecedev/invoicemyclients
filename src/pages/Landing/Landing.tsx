import React from 'react';

import { NavBar } from '../../components/NavBar';
import { Hero } from '../../components/Hero';
import { LandingPageDashboardPreview } from '../../components/LandingPageDashboardPreview';
import { Hub } from '../../components/Hub';
import { Pricing } from '../../components/Pricing';
import { LandingPageCTA } from '../../components/LandingPageCTA';
import { Footer } from '../../components/Footer';

const LandingPage = () => (
	<>
		<NavBar />
		<Hero />
		<LandingPageDashboardPreview />
		<Hub />
		<Pricing />
		<LandingPageCTA />
		<Footer />
	</>
);

export { LandingPage };
