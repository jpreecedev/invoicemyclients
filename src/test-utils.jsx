import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
