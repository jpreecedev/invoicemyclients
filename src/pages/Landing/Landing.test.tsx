import React from 'react'

import { LandingPage } from './Landing'
import { renderWithRouter } from '../../test-utils'

describe('<Landing /> tests', () => {
  it('should match snapshot', () => {
    expect(renderWithRouter(<LandingPage />).container).toMatchSnapshot()
  })
})
