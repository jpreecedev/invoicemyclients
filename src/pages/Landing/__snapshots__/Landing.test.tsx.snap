// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> tests should match snapshot 1`] = `
<div>
  <div>
    <div>
      <div>
        <div
          class="ui inverted vertical center aligned segment"
          style="min-height: 700px; padding: 1em 0em; background-image: url(landing-hero.jpg);"
        >
          <div
            class="ui large inverted pointing secondary menu"
          >
            <div
              class="ui container"
            >
              <a
                class="item"
              >
                Features
              </a>
              <a
                class="item"
              >
                Pricing
              </a>
              <div
                class="right item"
              >
                <a
                  class="ui primary button"
                  role="button"
                  style="margin-left: 0.5em;"
                >
                  Sign Up
                </a>
              </div>
            </div>
          </div>
          <div
            class="ui text container"
          >
            <h1
              class="ui inverted header"
              style="font-size: 4em; margin-bottom: 0px; margin-top: 3em;"
            >
              InvoiceMyClients.com
            </h1>
            <h2
              class="ui inverted header"
              style="font-size: 1.7em; margin-bottom: 2rem;"
            >
              We make it quick, easy, and secure for your clients to pay your invoices.
            </h2>
            <a
              class="ui huge primary button"
              href="/sign-up"
              role="button"
            >
              It's free to get started
              <i
                aria-hidden="true"
                class="arrow right icon"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        class="ui vertical segment"
        style="padding: 8em 0em;"
      >
        <div
          class="ui container stackable middle aligned grid"
        >
          <div
            class="row"
          >
            <div
              class="eight wide column"
            >
              <h3
                class="ui header"
                style="font-size: 2em;"
              >
                We Help Companies and Companions
              </h3>
              <p
                style="font-size: 1.33em;"
              >
                We can give your company superpowers to do things that they never thought possible. Let us delight your customers and empower your needs... through pure data analytics.
              </p>
              <h3
                class="ui header"
                style="font-size: 2em;"
              >
                We Make Bananas That Can Dance
              </h3>
              <p
                style="font-size: 1.33em;"
              >
                Yes that's right, you thought it was the stuff of dreams, but even bananas can be bioengineered.
              </p>
            </div>
            <div
              class="right floated six wide column"
            >
              <img
                class="ui large bordered rounded image"
                src="/images/wireframe/white-image.png"
              />
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="center aligned column"
            >
              <button
                class="ui huge button"
              >
                Check Them Out
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui vertical segment"
        style="padding: 0em;"
      >
        <div
          class="ui stackable internally celled equal width grid"
        >
          <div
            class="center aligned row"
          >
            <div
              class="column"
              style="padding-bottom: 5em; padding-top: 5em;"
            >
              <h3
                class="ui header"
                style="font-size: 2em;"
              >
                "What a Company"
              </h3>
              <p
                style="font-size: 1.33em;"
              >
                That is what they all say about us
              </p>
            </div>
            <div
              class="column"
              style="padding-bottom: 5em; padding-top: 5em;"
            >
              <h3
                class="ui header"
                style="font-size: 2em;"
              >
                "I shouldn't have gone with their competitor."
              </h3>
              <p
                style="font-size: 1.33em;"
              >
                <img
                  class="ui avatar image"
                  src="/images/avatar/large/nan.jpg"
                />
                <b>
                  Nan
                </b>
                 Chief Fun Officer Acme Toys
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ui vertical segment"
        style="padding: 8em 0em;"
      >
        <div
          class="ui text container"
        >
          <h3
            class="ui header"
            style="font-size: 2em;"
          >
            Breaking The Grid, Grabs Your Attention
          </h3>
          <p
            style="font-size: 1.33em;"
          >
            Instead of focusing on content creation and hard work, we have learned how to master the art of doing nothing by providing massive amounts of whitespace and generic content that can seem massive, monolithic and worth your attention.
          </p>
          <a
            class="ui large button"
            role="button"
          >
            Read More
          </a>
          <h4
            class="ui horizontal divider header"
            style="margin: 3em 0em; text-transform: uppercase;"
          >
            <a
              href="/case-studies"
            >
              Case Studies
            </a>
          </h4>
          <h3
            class="ui header"
            style="font-size: 2em;"
          >
            Did We Tell You About Our Bananas?
          </h3>
          <p
            style="font-size: 1.33em;"
          >
            Yes I know you probably disregarded the earlier boasts as non-sequitur filler content, but it's really true. It took years of gene splicing and combinatory DNA research, but our bananas can really dance.
          </p>
          <a
            class="ui large button"
            role="button"
          >
            I'm Still Quite Interested
          </a>
        </div>
      </div>
      <div
        class="ui inverted vertical segment"
        style="padding: 5em 0em;"
      >
        <div
          class="ui container"
        >
          <div
            class="ui inverted stackable divided grid"
          >
            <div
              class="row"
            >
              <div
                class="three wide column"
              >
                <h4
                  class="ui inverted header"
                >
                  About
                </h4>
                <div
                  class="ui inverted link list"
                  role="list"
                >
                  <a
                    class="item"
                    role="listitem"
                  >
                    Sitemap
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Contact Us
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Religious Ceremonies
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Gazebo Plans
                  </a>
                </div>
              </div>
              <div
                class="three wide column"
              >
                <h4
                  class="ui inverted header"
                >
                  Services
                </h4>
                <div
                  class="ui inverted link list"
                  role="list"
                >
                  <a
                    class="item"
                    role="listitem"
                  >
                    Banana Pre-Order
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    DNA FAQ
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    How To Access
                  </a>
                  <a
                    class="item"
                    role="listitem"
                  >
                    Favorite X-Men
                  </a>
                </div>
              </div>
              <div
                class="seven wide column"
              >
                <h4
                  class="ui inverted header"
                >
                  Footer Header
                </h4>
                <p>
                  Extra space for a call to action inside the footer that could help re-engage users.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
